djangocms_navigation/views.py

```diff
@@ -0,0 +1,98 @@
+from django.apps import apps
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponseBadRequest, JsonResponse
+from django.views.generic import View
+
+from cms.models import Page
+
+from djangocms_navigation.utils import is_model_supported, supported_models
+
+
+class ContentObjectSelect2View(View):
+    def get(self, request, *args, **kwargs):
+
+        content_type_id = self.request.GET.get("content_type_id", None)
+        # Return http bad request if there is no content_type_id provided in request
+        if not content_type_id:
+            return HttpResponseBadRequest()
+
+        # return http bad request if content_type not exist in db
+        try:
+            content_object = ContentType.objects.get_for_id(content_type_id)
+        except ContentType.DoesNotExist:
+            return HttpResponseBadRequest()
+
+        # return http request if content type is not registered to use navigation app
+        model = content_object.model_class()
+        if not is_model_supported(model):
+            return HttpResponseBadRequest()
+
+        self.site = get_current_site(request)
+        self.object_list = self.get_data()
+
+        data = {
+            "results": [{"text": str(obj), "id": obj.pk} for obj in self.get_data()]
+        }
+        return JsonResponse(data)
+
+    def dispatch(self, request, *args, **kwargs):
```

Comment:
Can't the whole `dispatch` override be removed now that the admin site decorator has been added to `get_urls`?